### PR TITLE
Add new H3 api method #h3ToNoChildrenIntersecting

### PR DESCRIPTION
**Pull request by iverase**
Given an H3 index, this method returns all the h3 index in the next resolution that intersects the provided h3 index but they are not children from it.

### CODE REVIEW

File: libs/h3/src/main/java/org/elasticsearch/h3/H3.java [hunk 1]
```diff
@@ -248,6 +248,34 @@ public static String[] h3ToChildren(String h3Address) {
...
+        Iterator.iterStepChild(it);
+        while (it.h != Iterator.H3_NULL) {
+            children[pos] = HexRing.h3NeighborInDirection(it.h, directions[pos++]);
+            Iterator.iterStepChild(it);
+        }
+        return children;
+    }
+
+    /**
+     * Transforms a list of H3 indexes in long form to a list of H3
```
Comment (craigtaverner) [inline libs/h3/src/main/java/org/elasticsearch/h3/H3.java:272]: The comment is not correct. It seems to be the description of the `h3ToStringList` function.

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java [hunk 2]
```diff
@@ -65,4 +72,38 @@ private void assertHexRing(int res, String h3Address, String[] children) {
             assertEquals(children[i], ring[positions[i - 1]]);
         }
     }
+
+    public void testNoChildrenIntersecting() {
+        String[] h3Addresses = H3.getStringRes0Cells();
+        String h3Address = RandomPicks.randomFrom(random(), h3Addresses);
+        for (int i = 1; i <= H3.MAX_H3_RES; i++) {
```
Comment (craigtaverner) [inline libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:79]: The for loop seems to imply the test will test at different resolutions, but it does not. It only selects 15 random level0 cells and tests those. I'm guessing you were planning something else, but did not complete it?

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java [hunk 3]
```diff
@@ -65,4 +72,38 @@ private void assertHexRing(int res, String h3Address, String[] children) {
...
+            h3Addresses = H3.h3ToChildren(h3Address);
+            assertIntersectingChildren(h3Address, h3Addresses);
+            h3Address = RandomPicks.randomFrom(random(), h3Addresses);
+        }
+    }
+
+    private void assertIntersectingChildren(String h3Address, String[] children) {
+        String[] intersectingChildren = H3.h3ToNoChildrenIntersecting(h3Address);
```
Comment (craigtaverner) [inline libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:87]: Perhaps the variable could be named `intersectingNotChildren`?

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java [hunk 4]
```diff
@@ -65,4 +72,38 @@ private void assertHexRing(int res, String h3Address, String[] children) {
...
+        for (String child : intersectingChildren) {
```
Comment (craigtaverner) [inline libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:88]: Perhaps the variable could be called `notChild`?

**Review by craigtaverner — CHANGES_REQUESTED** [review]
I love the function, but think there is insufficient test coverage since we only test level0 cells. I think we should test all resolutions.

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java (hunk 3 above)
Comment (iverase) [reply libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:87]: done

File: libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java (hunk 4 above)
Comment (iverase) [reply libs/h3/src/test/java/org/elasticsearch/h3/ParentChildNavigationTests.java:88]: done

File: libs/h3/src/main/java/org/elasticsearch/h3/H3.java (hunk 1 above)
Comment (iverase) [reply libs/h3/src/main/java/org/elasticsearch/h3/H3.java]: done

**Review by craigtaverner — APPROVED** [review]
Very useful addition!

**Pull request merged**